sorc/chgres_cube.fd/surface.F90

```diff
@@ -1202,8 +1207,13 @@ subroutine interp(localpet)
     if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
        call error_handler("IN FieldGather", rc)
 
-   call search_many(num_fields,bundle_landice_target,data_one_tile, land_target_one_tile,& 
-                    tile,search_nums,localpet,terrain_land=data_one_tile2,field_data_3d=data_one_tile_3d)
+   if (localpet==0) then
+      call search_many(num_fields,bundle_landice_target,tile,search_nums,localpet,&
+                terrain_land=data_one_tile2,mask=land_target_one_tile)
+   else
+      call search_many(num_fields,bundle_landice_target,tile,search_nums,localpet,&
```

Comment:
I think you can remove terrain_land from the argument list for all tasks > 0.

Reply:
Removed in my latest commit.